lib/index.ts

```diff
@@ -571,4 +574,40 @@ export default class WalletClient {
       this.eventListeners.get(event)!.forEach((listener) => listener(data));
     }
   }
+
+  private populateWindowObject() {
```

Comment:
Consider renaming to mutateWindow or patchWindow (or any relative naming that will describe that window are getting pathed and getting new methods in it
)